lib/generamba/cli/gen_command.rb

```diff
@@ -51,20 +51,18 @@ def gen(module_name, template_name)
 
       DependencyChecker.check_all_required_dependencies_has_in_podfile(template.dependencies, code_module.podfile_path)
 
-      project = XcodeprojHelper.obtain_project(code_module.xcodeproj_path)
-      module_group_already_exists = XcodeprojHelper.module_with_group_path_already_exists(project, code_module.module_group_path)
+      generator = Generamba::ModuleGenerator.new()
 
-      if module_group_already_exists
+      if generator.module_exist?(module_name, code_module, template)
```

Comment:
I don't like the idea of adding another one responsibility to a ModuleGenerator. It's already an object with a rather complex logic.

You may add another class - something like ModuleExistanceChecker and move this logic there